### ISSUE TITLE
Is this a drop-in replacement for https-proxy-agent?

### ISSUE THREAD

**Issue by zkochan** [delvedor/hpagent#106]
I am coming from [this issue](https://github.com/TooTallNate/node-https-proxy-agent/issues/89), where @dscalzi has suggested to used this package instead.

I use https-proxy-agent in [pnpm](https://github.com/pnpm/pnpm) and I wonder if it is safe to make the switch. I did try to switch it and everything seems to works. But I do see some difference between the two packages. Like I don't see in this package anything related to ALPNProtocols, like in https-proxy-agent:

https://github.com/TooTallNate/node-https-proxy-agent/blob/d0d80cc0482f20495aa8595f802e1a9f3b1b3409/src/agent.ts#L61-L65

Is this package a feature complete drop-in replacement to https-proxy-agent?

**Comment by trevor-rex** [comment]
Have you had success in using this package instead? It does seem slightly more up-to-date than TooTallNate's package. Do you happen to know if this is compatible with Node 16?

**Comment by zkochan** [comment]
No, I have not risked to make a switch. I have used a workaround to make https-proxy-agent work: https://github.com/pnpm/components/commit/02d17cc303c94dc7d85458fd233303edcc452c8b

**Comment by scarabcoder** [comment]
For future people finding this from Google, I found this to be a good replacement for https-proxy-agent. I was having issues with https-proxy-agent not passing the `ca` certificate to the CONNECT HTTP call in addition to the proxied request, which this library solved. 
It doesn't look like this library has been updated in awhile, but it is working for me.
